### PR TITLE
XMLENC from .NET Core.

### PR DESCRIPTION
**Pull request by vladimir-kazakov**
Currently, one test fails:
```
Constructor_X509CertificateByteArray_Null (MonoTests.System.Security.Cryptography.Xml.KeyInfoX509DataTest.Constructor_X509CertificateByteArray_Null)
   System.Security.Cryptography.CryptographicException : Certificate instance is empty.
  at System.Security.Cryptography.X509Certificates.X509Helper.ThrowIfContextInvalid (System.Security.Cryptography.X509Certificates.X509CertificateImpl impl) [0x0000e] in /mono/mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs:161 
  at System.Security.Cryptography.X509Certificates.X509Certificate.get_Impl () [0x00000] in /mono/mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Certificate.cs:137 
  at Mono.Btls.MonoBtlsProvider.GetNativeCertificate (System.Security.Cryptography.X509Certificates.X509Certificate certificate) [0x00000] in /mono/mcs/class/System/Mono.Btls/MonoBtlsProvider.cs:112 
  at System.Security.Cryptography.X509Certificates.X509Helper2.Import (System.Security.Cryptography.X509Certificates.X509Certificate cert, System.Boolean disableProvider) [0x00011] in /mono/mcs/class/System/System.Security.Cryptography.X509Certificates/X509Helper2.cs:149 
  at System.Security.Cryptography.X509Certificates.X509Certificate2..ctor (System.Security.Cryptography.X509Certificates.X509Certificate certificate) [0x0000b] in /mono/mcs/class/System/System.Security.Cryptography.X509Certificates/X509Certificate2.cs:134 
  at System.Security.Cryptography.Xml.KeyInfoX509Data.AddCertificate (System.Security.Cryptography.X509Certificates.X509Certificate certificate) [0x00021] in /mono/external/corefx/src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs:116 
  at System.Security.Cryptography.Xml.KeyInfoX509Data..ctor (System.Byte[] rgbCert) [0x0000d] in /mono/external/corefx/src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs:39 
  at MonoTests.System.Security.Cryptography.Xml.KeyInfoX509DataTest.Constructor_X509CertificateByteArray_Null () [0x00000] in /mono/mcs/class/System.Security/Test/System.Security.Cryptography.Xml/KeyInfoX509DataTest.cs:90 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in /mono/mcs/class/corlib/System.Reflection/MonoMethod.cs:305
```

This issue is related to Mono and can be reproduced by executing the following line of code:
```
new X509Certificate2(new X509Certificate2());
```

In Mono it throws (which is wrong), but in .NET Framework and .NET Core it doesn't (which is the correct behavior, according to docs: https://msdn.microsoft.com/en-us/library/ms223098(v=vs.110).aspx).

I'm currently looking into the implementation of X.509 certificates, but due to its complexity, it may take some time until I'll find what exactly is wrong and how to fix it. Perhaps someone from maintainers may somehow help?

@baulig, I saw your commits related to BTLS. Looks like it's somehow related. Any thoughts regarding this issue?

### CODE REVIEW

File: mcs/class/System.Security/Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ MONO_SECURITY=Mono.Security
 endif
 
 LIBRARY = System.Security.dll
-LIB_REFS = secxml/System bare/System.Xml $(MONO_SECURITY)
+LIB_REFS = secxml/System bare/System.Xml System.Numerics $(MONO_SECURITY)
```
Comment (marek-safar) [inline mcs/class/System.Security/Makefile]: You cannot do that, System.Numerics is built after System.Security. What code does require it?

Comment (vladimir-kazakov) [inline mcs/class/System.Security/Makefile]: The `KeyInfoX509Data` uses `BigInteger` from there: https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs#L164.

**Comment by vladimir-kazakov** [comment]
The last commit reverts changes related to `KeyInfoX509Data`, since it depends on `BigInteger` from the `System.Numerics` assembly. Currently, I don't know how to solve this problem, so I'd like to forget about this problem for now and focus on something more important. Perhaps in the future it will be more obvious how to address this problem.

Speaking of failed tests, seems like they're not related to changes in this PR, except changes in the public API. But, if I understand it correctly, such changes should be committed by a maintainer.

**Comment by marek-safar** [comment]
Thanks for the contribution. I agree it's easiest to revert KeyInfoX509Data changes for now.

API Diff shows 2 removed methods
```
public EncryptedXml(System.Xml.XmlDocument document, System.Security.Policy.Evidence evidence)          
public System.Security.Policy.Evidence DocumentEvidence { get; set }
```

which we need to bring back for full .net framework compatibility

**Comment by vladimir-kazakov** [comment]
@marek-safar, missing members were added to .NET Core, and should be available after updating the `corefx` submodule.

**Comment by marek-safar** [comment]
@vladimir-kazakov awesome, we are working on updating corefx but it's tricky this time

**Comment by marek-safar** [comment]
build

**Comment by vladimir-kazakov** [comment]
Rebased on the top of the master.

**Comment by marek-safar** [comment]
commit apidiff

**Comment by marek-safar** [comment]
@vladimir-kazakov Thanks for the PR. I appreciate your patience.

**Pull request merged**